chains/linearizable_vm.go

```diff
@@ -28,26 +28,41 @@ type initializeOnLinearizeVM struct {
 	vmToInitialize common.VM
 	vmToLinearize  *linearizeOnInitializeVM
 
-	ctx          *snow.Context
-	db           database.Database
-	genesisBytes []byte
-	upgradeBytes []byte
-	configBytes  []byte
-	toEngine     chan<- common.Message
-	fxs          []*common.Fx
-	appSender    common.AppSender
+	ctx              *snow.Context
+	db               database.Database
+	genesisBytes     []byte
+	upgradeBytes     []byte
+	configBytes      []byte
+	fxs              []*common.Fx
+	appSender        common.AppSender
+	waitForLinearize chan struct{}
+}
+
+func (vm *initializeOnLinearizeVM) WaitForEvent(ctx context.Context) (common.Message, error) {
+	select {
+	case <-vm.waitForLinearize:
+		return vm.vmToInitialize.WaitForEvent(ctx)
+	case <-ctx.Done():
+		return 0, ctx.Err()
+	}
 }
 
 func (vm *initializeOnLinearizeVM) Linearize(ctx context.Context, stopVertexID ids.ID) error {
 	vm.vmToLinearize.stopVertexID = stopVertexID
+	select {
+	case <-vm.waitForLinearize:
+	default:
+		defer func() {
+			close(vm.waitForLinearize)
+		}()
+	}
```

Comment:
While this does work to ensure that the channel is only closed once... It feels odd. Would it be cleaner to use a `sync.Once`? Or honestly even panic if we call `Linearize` multiple times.